messaging/src/main/java/org/axonframework/eventhandling/MultiSourceTrackingToken.java

```diff
@@ -0,0 +1,194 @@
+
+    private static final Logger logger = LoggerFactory.getLogger(MultiSourceTrackingToken.class);
+
+    @JsonTypeInfo(use = JsonTypeInfo.Id.MINIMAL_CLASS)
+    private final Map<String, TrackingToken> trackingTokens;
+
+    /**
+     * Construct a new {@link MultiSourceTrackingToken} from a map of existing tokens.
+     *
+     * @param trackingTokens the map of tokens which make up the {@link MultiSourceTrackingToken}
+     */
+    @JsonCreator
+    public MultiSourceTrackingToken(@JsonProperty("trackingTokens") Map<String, TrackingToken> trackingTokens) {
+        this.trackingTokens = trackingTokens;
+    }
+
+
+    /**
+     * Compares this token to {@code other} by comparing each member token with its counterpart in the {@code other}
+     * token
+     *
+     * @param other The token to compare to this one
+     * @return token representing the lower bound of of both tokens
+     */
+    @Override
+    public TrackingToken lowerBound(TrackingToken other) {
+        Assert.isTrue(other instanceof MultiSourceTrackingToken, () -> "Incompatible token type provided.");
+
+        MultiSourceTrackingToken otherMultiToken = (MultiSourceTrackingToken) other;
+
+        Assert.isTrue(otherMultiToken.trackingTokens.keySet().equals(this.trackingTokens.keySet()),
+                      () -> "MultiSourceTrackingTokens contain different keys");
+
+        Map<String, TrackingToken> tokenMap = new HashMap<>();
+
+        otherMultiToken.trackingTokens.forEach((key, otherToken) ->
+                                                       tokenMap.put(key, trackingTokens.get(key).lowerBound(otherToken))
+        );
+
+        return new MultiSourceTrackingToken(tokenMap);
+    }
+
+    /**
+     * Compares this token to {@code other} by comparing each member token with its counterpart in the {@code other}
+     * token
+     *
+     * @param other The token to compare this token to
+     * @return a token that represents the furthest position of this or the other streams
+     */
+    @Override
+    public TrackingToken upperBound(TrackingToken other) {
+        Assert.isTrue(other instanceof MultiSourceTrackingToken, () -> "Incompatible token type provided.");
+
+        MultiSourceTrackingToken otherMultiToken = (MultiSourceTrackingToken) other;
+
+        Assert.isTrue(otherMultiToken.trackingTokens.keySet().equals(this.trackingTokens.keySet()),
+                      () -> "MultiSourceTrackingTokens contain different keys");
+
+        Map<String, TrackingToken> tokenMap = new HashMap<>();
+
+        otherMultiToken.trackingTokens.forEach((key, otherToken) ->
+                                                       tokenMap.put(key, trackingTokens.get(key).upperBound(otherToken))
+        );
+
+        return new MultiSourceTrackingToken(tokenMap);
+    }
+
+    /**
+     * Compares this token to {@code other} checking each member token with its counterpart to see if they are covered
```

Comment:
I think it would be beneficial if the JavaDoc clearly states that the used tokens for this operation have a similar set of message source names, as otherwise they cannot originate from the same `MultiStreamableMessageSource`.